rust_src/src/strings.rs

```diff
@@ -202,6 +202,43 @@ fn string_to_unibyte(string: LispObject) -> LispObject {
     }
 }
 
+fn get_string_or_symbol(mut string: LispObject) -> multibyte::LispStringRef {
+    if string.is_symbol() {
+        string = string.symbol_name()
+    }
+
+    string.as_string_or_error()
+}
+
+fn string_lessp(string1: LispObject, string2: LispObject) -> LispObject {
+    let mut lispstr1 = get_string_or_symbol(string1);
+    let mut lispstr2 = get_string_or_symbol(string2);
+
+    let end = cmp::min(lispstr1.len_bytes(), lispstr2.len_bytes());
+    let mut i1 = 0;
+    while i1 < end {
+        // Unwraps should be fine here, due to our manual tracking of
```

Comment:
hmm, could this be done with a `zip`? I don't think we need `i1` afterwards since if we finish the loop it will always be `= end`.